### PR TITLE
document generator consumer update to remove resourceID

### PR DESCRIPTION
**Pull request by Mathsh**
update to document generator consumer to no longer use the resourceId as this will be removed from the message received from Kafka.

- Removed reference to resourceId 
- Updated the setRequest in DocumentGeneratorImpl to add the authorisation header to the rest request.
- Added the Api_Key to the parameter check
- Altered the @autowired services to use constructor injection
- Updated the debug maps in MessageProcessorRunnerImpl to add the Kafka details such as offset, etc.
- Added static final variables for map params
- Added log.infocontext message when the consumer has completed the generation of the document

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/document/generator/consumer/document/service/impl/GenerateDocumentImpl.java [hunk 1]
```diff
@@ -22,38 +26,58 @@
...
         } catch (Exception e) {
-           throw new GenerateDocumentException("An error occurred when requesting the generation" +
+            LOG.errorContext("Error occurred during api call to document-generator",
+                    e, setDebugMap(deserialisedKafkaMessage));
+            throw new GenerateDocumentException("An error occurred when requesting the generation" +
                    " of a document from the document generator api", e);
         }
     }
 
+    private Object setRequest(DeserialisedKafkaMessage deserialisedKafkaMessage) {
```
Comment (sroberts2) [inline src/main/java/uk/gov/companieshouse/document/generator/consumer/document/service/impl/GenerateDocumentImpl.java]: Can we return something more specific than Object.

File: src/main/java/uk/gov/companieshouse/document/generator/consumer/processor/impl/MessageProcessorImpl.java [hunk 2]
```diff
@@ -28,23 +28,37 @@
...
+                                AvroDeserializer<DeserialisedKafkaMessage> avroDeserializer) {
+
+        this.messageService = messageService;
+        this.generateDocument = generateDocument;
+        this.kafkaConsumerService = kafkaConsumerService;
+        this.kafkaProducerService = kafkaProducerService;
+        this.avroDeserializer = avroDeserializer;
+    }
+
+    private static final String KAFKA_MSG = "kafka_message";
```
Comment (sroberts2) [inline src/main/java/uk/gov/companieshouse/document/generator/consumer/processor/impl/MessageProcessorImpl.java]: Field declaration should appear before constructors.

File: src/main/java/uk/gov/companieshouse/document/generator/consumer/document/service/impl/GenerateDocumentImpl.java (hunk 1 above)
Comment (Mathsh) [reply src/main/java/uk/gov/companieshouse/document/generator/consumer/document/service/impl/GenerateDocumentImpl.java]: done, updated to HttpEntity<GenerateDocumentRequest> as that is what it is actually returning

File: src/main/java/uk/gov/companieshouse/document/generator/consumer/processor/impl/MessageProcessorImpl.java (hunk 2 above)
Comment (Mathsh) [reply src/main/java/uk/gov/companieshouse/document/generator/consumer/processor/impl/MessageProcessorImpl.java]: done

**Pull request merged**